xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererGL.cpp

```diff
@@ -89,10 +106,8 @@ static void LoadTexture(GLenum target
     stride = width;
   }
   /** OpenGL ES does not support strided texture input. Make a copy without stride **/
-  else if (stride != width)
+  else if (stride != bytesPerLine)
```

Comment:
This seems to change the behaviour. Before it we compared stride to width, now you are comparing stride to stride, why?

Reply:
`int bytesPerLine = bytesPerPixel * width`, so this is still comparing stride to width. If you look at previous implementation, it wasn't entirely correct because it compared stride (in BYTES) to width (in PIXELS) - but it was okay since this comparison was only made if `alpha == true`, i.e. one byte per pixel. Now we need to take bytes per pixel into consideration. 

In the Avatar sample stride = 7680, width = 1920, bpp = 4, so we really need to compare stride to bytesPerLine.